src/client/updaters/tag.js

```diff
@@ -0,0 +1,84 @@
+// borrow the slice method
+const toArray = Function.prototype.call.bind(Array.prototype.slice)
+
+/**
+ * Updates meta tags inside <head> and <body> on the client. Borrowed from `react-helmet`:
+ * https://github.com/nfl/react-helmet/blob/004d448f8de5f823d10f838b02317521180f34da/src/Helmet.js#L195-L245
+ *
+ * @param  {('meta'|'base'|'link'|'style'|'script'|'noscript')} type - the name of the tag
+ * @param  {(Array<Object>|Object)} tags - an array of tag objects or a single object in case of base
+ * @return {Object} - a representation of what tags changed
+ */
+export default function updateTag({ attribute, tagIDKeyName } = {}, type, tags, headTag, bodyTag) {
+  const oldHeadTags = toArray(headTag.querySelectorAll(`${type}[${attribute}]`))
+  const oldBodyTags = toArray(bodyTag.querySelectorAll(`${type}[${attribute}][data-body="true"]`))
+  const newTags = []
+
+  if (tags.length > 1) {
+    // remove duplicates that could have been found by merging tags
+    // which include a mixin with metaInfo and that mixin is used
+    // by multiple components on the same page
+    const found = []
+    tags = tags.filter((x) => {
+      const k = JSON.stringify(x)
+      const res = !found.includes(k)
+      found.push(k)
```

Comment:
Do we *have to* push "duplicate tags"?

Reply:
To be honest, I dont know. Or in other words, after reading the description of why this code exists I couldnt be bothered enough (for now) to write a test for it (and there where other areas where more/greater improvements could be made)